src/control/conf.c

```diff
@@ -125,7 +130,28 @@ int dt_conf_get_int(const char *name)
 {
   const char *str = dt_conf_get_var(name);
   float new_value = dt_calculator_solve(1, str);
-  if(isnan(new_value)) new_value = 0.0;
+  if(isnan(new_value))
```

Comment:
If we have no garbage and there is a min/max set in conf, can we clamp the value to ensure it is still in the right range? This would add another layer of safety.

Reply:
I did that already in #6364 by introducing `dt_conf_get_and_sanitize_*` set of functions. This puts a burden of clamping on code calling and adds very little overhead IMO. Otherwise conf.c would have to also load + parse xml with min/max/default which could be a bit over the top imo... I mean... It's possible, but not really needed in many cases, and cases that do need clamped values ca use `dt_conf_get_and_sanitize_*` class of calls :)

Reply:
TL;DR my explaination: me to dum to do dat ;(

Reply:
Indeed, we don't have code generated for min/max in preferences.

Reply:
> Indeed, we don't have code generated for min/max in preferences.

Haven't looked at all yet how all of this works, but rather than live reading the xml at run-time, could some validation code be generated that fixes up all values (clamping plus defaulting for garbage) in conf.c at load time? Once the whole darktablerc is read, it stays in memory so won't need to be revalidated again.

This doesn't yet cover the case where the code would want to use the min/max to do clamping on values that are generated some other way (i.e. not read from the config). It would be a bit unfortunate to have to maintain min/max both in the xml _and_ in the code (since they could easily get out of sync). Maybe again a statically generated lookup table, the code for which would be generated at compile time?

Reply:
another `generate_prefs.xsl` only for default config?  huh... that might be super painful...

Reply:
Isn't there already something like that in case there is no darktablerc at all yet? I believe a new one, containing all the default values, is auto generated, presumably based on the xml, no? Don't know how any of this works. "xsl"! Brrr.

Reply:
the "default" mechanism relies on existence of "darktablerc" in darktable install datadir :) if that's missing then... it might not play nice :)

Reply:
Ah. And that does not get autogenerated from the xml?

Generating validation code should be somewhat simpler than generating the preference dialog, no? But it has been a very long time since I looked at xsl. And that was from a long distance and between my fingers...

Reply:
What would be needed is an xsl that generates just a long list of calls to 
dt_conf_get_and_sanitize_,  filling in the min/max values from darktableconfig.xml. This code would then be called from conf.c at startup. 
Actually, it would be better to extend "sanitize" so that it would fill in the default value, instead of the max, when there is no value at all.
With that, there would be no need to sprinkle calls to dt_conf_get_and_sanitize_ all around the code (and there would only be _one_ authoritative definition of the min/max values.

Reply:
That would work for predefined conf vals that are defined in darktableconfig.xml.in... some are dynamic etc. But sorta yeah, agree on generating some sort of safe conf

Reply:
I'm working on a xsl to generate min/max/default and corresponding API.

Reply:
I have some ideas but I do trust you to come up with  something better :)

Reply:
> I'm working on a xsl to generate min/max/default and corresponding API.

Please make it so it doesn't require changes everywhere to benefit from it (i.e. the default "get" functions should receive valid/cleaned responses. Load time would be the correct time to validate the values in darktablerc once and for all, imho.

But would be great to be able to access min, max, default separately from value via an api afterwards.

Reply:
WIP here : #6412